crates/typst/src/geom/color.rs

```diff
@@ -27,68 +115,421 @@ use crate::syntax::Spanned;
 /// #rect(fill: aqua)
 /// #rect(fill: color.aqua)
 /// ```
-#[ty(scope)]
 #[derive(Copy, Clone, Eq, PartialEq, Hash)]
+#[ty(scope)]
```

Comment:
With `Color` now being `40` bytes large, it increases the size of `Value`, which is very unfortunate. I assume f32 is not enough? Otherwise, perhaps we need to wrap it in `Arc`, i.e. `Color(Arc<Repr>)` + `enum Repr { .. }`.

Reply:
I am going back to f32s, over 200 compile errors 😱